cl/corpus_importer/management/commands/update_opinions_order.py

```diff
@@ -0,0 +1,96 @@
+from django.db.models import Count, Q
+
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.search.models import SOURCES, Opinion, OpinionCluster
+
+
+def sort_harvard_opinions(options) -> None:
+    """Sort harvard opinions
+
+    We assume that harvard data is already ordered, we just need to fill
+    the order field in each opinion
+
+    The harvard importer created the opinions in order of appearance in the file
+
+    :param options: dict of arguments skip until and limit if given
+    :return: None
+    """
+
+    skip_until = options.get("skip_until", None)
+    limit = options.get("limit", None)
+
+    base_filter = (
+        OpinionCluster.objects.exclude(filepath_json_harvard="")
+        .annotate(opinions_count=Count("sub_opinions"))
+        .filter(opinions_count__gt=1)
+    )
+
+    if skip_until:
+        base_filter &= Q(pk__gte=skip_until)
+
+    harvard_clusters = (
+        OpinionCluster.objects.annotate(opinions_count=Count("sub_opinions"))
+        .filter(base_filter)
+        .order_by("id")
+    )
+    if limit:
+        harvard_clusters = harvard_clusters[:limit]
+
+    for cluster in harvard_clusters:
+        logger.info(f"Processing cluster id: {cluster}")
+        sub_opinions = cluster.sub_opinions.exclude(
+            type=Opinion.COMBINED,
+        ).order_by("id")
+        if not sub_opinions:
+            logger.info(
+                f"No sub_opinions left to order for cluster id: {cluster}"
+            )
+            continue
```

Comment:
Instead of doing this query millions of times, could you do a check inside your for loop below and `continue` when you run into `Opinion.COMBINED`? You'd have to fiddle with the `opinion_order` value, but I think it could work, and save our poor DB a bit.